### PR TITLE
feat(eigen-services-operatorsinfo): Testing 

### PR DESCRIPTION
**Pull request by supernovahs**

### CODE REVIEW

File: crates/chainio/clients/avsregistry/src/reader.rs [hunk 1]
```diff
@@ -355,71 +353,77 @@ impl AvsRegistryChainReader {
...
-    /// Queies existing operators from for a particular block range
     /// Queies existing operators from for a particular block range
     pub async fn query_existing_registered_operator_pub_keys(
         &self,
         start_block: u64,
         mut stop_block: u64,
+        ws_url: String,
     ) -> Result<(Vec<Address>, Vec<OperatorPubKeys>), AvsRegistryError> {
-        let provider = get_provider(&self.provider);
+        // todo remove unwrap
```
Comment (dralves) [inline crates/chainio/clients/avsregistry/src/reader.rs:363]: there's no unwrap anymore, right

File: crates/chainio/clients/avsregistry/src/reader.rs [hunk 2]
```diff
@@ -355,71 +353,77 @@ impl AvsRegistryChainReader {
...
+                        let mut operator_addresses: Vec<Address> = vec![];
+                        let mut operator_pub_keys: Vec<OperatorPubKeys> = vec![];
+                        while i <= stop_block {
+                            let mut to_block = i + (query_block_range - 1);
+                            if to_block > stop_block {
+                                to_block = stop_block;
+                            }
+                            let filter = Filter::new()
+                                .select(i..to_block)
+                                .event("NewPubkeyRegistration(address,(uint256,uint256),(uint256[2],uint256[2]))")
```
Comment (dralves) [inline crates/chainio/clients/avsregistry/src/reader.rs:386]: pull a constant with proper naming

File: crates/chainio/clients/avsregistry/src/reader.rs [hunk 3]
```diff
@@ -355,71 +353,77 @@ impl AvsRegistryChainReader {
...
+                                .address(self.bls_apk_registry_addr);
+
+                            let logs_result = provider.get_logs(&filter).await;
+
+                            match logs_result {
+                                Ok(logs) => {
+                                    debug!(transactionLogs = ?logs, "avsRegistryChainReader.QueryExistingRegisteredOperatorPubKeys");
+
+                                    for v_log in logs.iter() {
+                                        if let Ok(pub_key_reg) =
```
Comment (dralves) [inline crates/chainio/clients/avsregistry/src/reader.rs:396]: what happens on Err, should we ignore or log? at least leave a comment

if we're ignore Err then you can do:
```
for v_log in logs.iter().map(|v| v.log_decode::<BLSApkRegistry::NewPubkeyRegistration>()).filter(|v| v.is_some())
```

and avoid the inner if

File: crates/chainio/clients/avsregistry/src/reader.rs [hunk 4]
```diff
@@ -355,71 +353,77 @@ impl AvsRegistryChainReader {
...
+                match current_block_number_result {
+                    Ok(current_block_number) => {
+                        if stop_block.is_zero() {
+                            stop_block = current_block_number;
+                        }
+                        let mut i = start_block;
+                        let mut operator_addresses: Vec<Address> = vec![];
+                        let mut operator_pub_keys: Vec<OperatorPubKeys> = vec![];
+                        while i <= stop_block {
+                            let mut to_block = i + (query_block_range - 1);
```
Comment (dralves) [inline crates/chainio/clients/avsregistry/src/reader.rs:380]:
```suggestion
                            let to_block = std::min(i + (query_block_range - 1), stop_block);
```

File: crates/services/operatorsinfo/src/operatorsinfo_inmemory.rs [hunk 5]
```diff
@@ -84,30 +88,41 @@ impl OperatorInfoServiceInMemory {
...
+                            X: event_data.pubkeyG1.X,
+                            Y: event_data.pubkeyG1.Y,
+                        },
+                        g2_pub_key: G2Point {
+                            X: event_data.pubkeyG2.X,
+                            Y: event_data.pubkeyG2.Y,
+                        },
+                    };
+                    // Send message
+                    let _ = pub_keys.send(OperatorsInfoMessage::InsertOperatorInfo(
```
Comment (dralves) [inline crates/services/operatorsinfo/src/operatorsinfo_inmemory.rs:118]: should we log here?

Comment (supernovahs) [reply crates/services/operatorsinfo/src/operatorsinfo_inmemory.rs:122]: Added a TODO to log after this gets merged, will do in #62

**Comment by supernovahs** [comment]
@anupsv  I am deploying on testing environment, but getting error `HOLESKY_WS_URL` not found . Can you check thanks

File: crates/services/operatorsinfo/src/operatorsinfo_inmemory.rs (hunk 5 above)
Comment (supernovahs) [reply crates/services/operatorsinfo/src/operatorsinfo_inmemory.rs:136]: added loggiing

**Comment by supernovahs** [comment]
@dralves

**Review by dralves — APPROVED** [review]

**Pull request merged**